### PR TITLE
Fix KernelSU

### PR DESCRIPTION
**Pull request by s1204IT**
```
windows_x86_64:/ $ /system/etc/preinstall.sh
Success
Starting: Intent { cmp=me.weishu.kernelsu/.ui.MainActivity }
windows_x86_64:/ $ /system/etc/preinstall.sh
```

Just one step away.

### CODE REVIEW

**Comment by s1204IT** [comment]
Now that we have finally confirmed that the script is working properly, all we need to do is to start it automatically.

**Comment by s1204IT** [comment]
🤔 
```
I IntentRedirectionHandler: Start activity for package me.weishu.kernelsu activityId 81800e3e-5d1e-41e9-b4d4-22a76f1130d5
I vol.VolumeDialogImpl: mDialog.dismiss() reason: volume_controller from: com.android.systemui.volume.VolumeDialogImpl$7.onDismissRequested:2074
E IntentRedirectionHandler: Failed to find intent for package launch: me.weishu.kernelsu
```

**Comment by YT-Advanced** [comment]
So can I merge it now ?

File: scripts/build.sh [hunk 1]
```diff
@@ -533,18 +533,9 @@ EOF
...
-    KSU_INIT="$SYSTEM_MNT/etc/init/hw/init.rc"
-    sudo tee "$KSU_INIT" <<EOF >/dev/null
-service ksuinstall logwrapper /system/etc/ksuinstall.sh
-    user shell
-    group shell
-    class late_start
-    seclabel u:r:shell:s0
-    disabled
-    oneshot
+    sudo tee "$$YSTEM_MNT/etc/init/hw/init.rc" <<EOF >/dev/null
```
Comment (YT-Advanced) [inline scripts/build.sh:536]: Double $$ here @s1204IT

Comment (s1204IT) [reply scripts/build.sh:536]: Oh, I was careless.

File: scripts/build.sh [hunk 2]
```diff
@@ -540,10 +540,10 @@ EOF
     sudo chown root:root "$KSU_PRE"
     sudo setfattr -n security.selinux -v "u:object_r:system_file:s0" "$KSU_PRE" || abort
     # Setup init
-#    sudo tee "$SYSTEM_MNT/etc/init/hw/init.rc" <<EOF >/dev/null
-#on property:sys.boot_completed=1
-#    exec - system system -- logwrapper /system/etc/ksuinstall.sh
-#EOF
+    sudo tee "$SYSTEM_MNT/etc/init/hw/init.rc" <<EOF >/dev/null
+on property:sys.boot_completed=1
+    exec u:r:init:s0 -- /system/bin/logwrapper /system/bin/sh /system/etc/ksuinstall.sh
```
Comment (s1204IT) [inline scripts/build.sh]: @YT-Advanced 
After all, this seems to be the reason why it will not start.
Even logcat does not work.

Comment (s1204IT) [reply scripts/build.sh]: Assumed behavior:
```
PS C:\Users\Administrator> adb shell /system/etc/ksuinstall.sh
Checking boot completed
Checking past install
Installing KernelSU APK
Success
Launching KernelSU App
Starting: Intent { cmp=me.weishu.kernelsu/.ui.MainActivity }
Placing completed file
Done!
```

Comment (s1204IT) [reply scripts/build.sh]: ohhh
My mistake completely.

Comment (YT-Advanced) [reply scripts/build.sh]: Oh, and now it worked

Comment (s1204IT) [reply scripts/build.sh]: In the meantime, I will try the multiple patterns I was going to try earlier.

File: scripts/build.sh [hunk 3]
```diff
@@ -527,6 +527,7 @@ echo "Checking past install"
 if [ ! -e "/storage/emulated/0/.ksu_completed_\$(getprop ro.build.date.utc)" ]; then
     echo "Installing KernelSU APK"
     pm install -i android -r /system/data-app/KernelSU.apk
+    appops set me.weishu.kernelsu ACCESS_RESTRICTED_SETTINGS allow
```
Comment (YT-Advanced) [inline scripts/build.sh:530]: no need it. Install with `-g` argument will grant all requested permissions

Comment (s1204IT) [reply scripts/build.sh:530]: Yes, of course I know.
I just tried to set the status of AppOps.

No need to do anything else.
https://github.com/tiann/KernelSU/blob/v0.5.7/manager/app/src/main/AndroidManifest.xml

I'll delete it later.

Comment (YT-Advanced) [reply scripts/build.sh]: @s1204IT Can you send the log ?

**Comment by YT-Advanced** [comment]
image

Use ksuinstall file and we have this error

**Comment by s1204IT** [comment]
> image
> 
> Use ksuinstall file and we have this error. Now I'm thinking about add auto install script to VPNDiaglogs apk


https://github.com/YT-Advanced/WSA-Script/assets/52069677/5ee887e2-ab0c-4f40-990d-0014428c2249

Does not work?

File: scripts/build.sh [hunk 4]
```diff
@@ -506,6 +497,64 @@ find "../common/system/priv-app/" -maxdepth 1 -mindepth 1 -printf '%P\n' | xargs
...
+    echo "Add auto-install for KernelSU Manager"
+    # Copy APK
+    DAT_APP="$SYSTEM_MNT/data-app"
+    sudo mkdir "$DAT_APP"
+    sudo cp "$KERNELSU_APK_PATH" "$DAT_APP/"
+    sudo chmod 0755 "$DAT_APP/"
+    sudo chmod 0644 "$DAT_APP/KernelSU.apk"
+    sudo find "$DAT_APP/" -exec chown root:root {} \;
+    sudo find "$DAT_APP/" -exec setfattr -n security.selinux -v "u:object_r:system_file:s0" {} \; || abort
+    # Setup script
```
Comment (YT-Advanced) [inline scripts/build.sh:514]: @hmtheboy154 
Vietnamese: Bác có cao kiến gì không ạ. Em thử nãy giờ vẫn chưa làm được script auto-install cái KernelSU
English: Do you have any idea ?

**Comment by YT-Advanced** [comment]
@s1204IT The KernelSU app worked correctly after restart WSA. So I will place KernelSU apk in `/system/app`

**Comment by s1204IT** [comment]
> @s1204IT The KernelSU app worked correctly after restart WSA. So I will place KernelSU apk in `/system/app`

Do you mean placing the APK in /system/app/KernelSU/, starting WSA once, shutting it down, and running it again?

**Comment by YT-Advanced** [comment]
> Do you mean placing the APK in /system/app/KernelSU/, starting WSA once, shutting it down, and running it again?
If so, remember, as the KernelSU developers have stated, it is essentially a user app, not a system app.

I knew. But if we have KernelSU, we can easily uninstall itself using adb with su access

**Comment by YT-Advanced** [comment]
> Do you mean placing the APK in /system/app/KernelSU/, starting WSA once, shutting it down, and running it again?

No, I meant that the old error disappear after restart. Now I'm trying place in `/system/app/` and get this error. Do you have any idea ?
```powershell
05-28 13:22:22.566   295   295 W ziparchive: Unable to open '/system/app/KernelSU/KernelSU.dm': No such file or directory

**Comment by s1204IT** [comment]
> I knew. But if we have KernelSU, we can easily uninstall itself using adb with su access

Fundamentally, when using ADB, there is no advantage to placing an APK on the system.

I think we can give up on automatic installation by the system now and change to installing `ksuinstall` commands directly from ADB, as I've created.

**Comment by YT-Advanced** [comment]
> I think we can give up on automatic installation by the system now and change to installing `ksuinstall` commands directly from ADB, as I've created.

Now it done, just need to fix permission of WSA's lib

**Comment by s1204IT** [comment]
Just to confirm, do we just put the library and APK in /system/app/KernelSU/, or do we just need to solve the problem of missing library permissions?

script and init.rc not used?

---
※The text may not be correct because it may not have been translated well.

**Comment by YT-Advanced** [comment]
> script and init.rc not used?

Yes, I place apk and lib in /system/app, and create Android.mk to import lib

**Comment by YT-Advanced** [comment]
@s1204IT Can you tell me which commit you are using in your video. I think we will give up the auto install

**Comment by s1204IT** [comment]
> @s1204IT Can you tell me which commit you are using in your video. I think we will give up the auto install

It is my last build.
The SHA of the commit is also listed at the same time.
You may give up automatic installation.

**Pull request closed without merging**